src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -110,77 +78,78 @@ public class TxPool : ITxPool, IDisposable
         /// <param name="chainHeadInfoProvider"></param>
         /// <param name="txPoolConfig"></param>
-        /// <param name="stateProvider"></param>
-        /// <param name="transactionComparerProvider"></param>
         /// <param name="validator"></param>
         /// <param name="logManager"></param>
         /// <param name="comparer"></param>
-        public TxPool(ITxStorage txStorage,
+        /// <param name="incomingTxFilter"></param>
+        public TxPool(
+            ITxStorage txStorage,
             IEthereumEcdsa ecdsa,
             IChainHeadInfoProvider chainHeadInfoProvider,
             ITxPoolConfig txPoolConfig,
             ITxValidator validator,
             ILogManager? logManager,
-            IComparer<Transaction> comparer)
+            IComparer<Transaction> comparer,
+            IIncomingTxFilter? incomingTxFilter = null)
         {
-            _ecdsa = ecdsa ?? throw new ArgumentNullException(nameof(ecdsa));
-            _chainHeadInfoProvider = chainHeadInfoProvider ?? throw new ArgumentNullException(nameof(chainHeadInfoProvider));
             _logger = logManager?.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
+            _headInfo = chainHeadInfoProvider ?? throw new ArgumentNullException(nameof(chainHeadInfoProvider));
             _txStorage = txStorage ?? throw new ArgumentNullException(nameof(txStorage));
-            _specProvider = _chainHeadInfoProvider.SpecProvider;
-            _txPoolConfig = txPoolConfig;
-            _stateProvider = _chainHeadInfoProvider.ReadOnlyStateProvider;
-            _validator = validator ?? throw new ArgumentNullException(nameof(validator));
+            _accounts = _headInfo.AccountStateProvider;
+            _specProvider = _headInfo.SpecProvider;
 
             MemoryAllowance.MemPoolSize = txPoolConfig.Size;
-            ThisNodeInfo.AddInfo("Mem est tx   :",
-                $"{(LruCache<Keccak, object>.CalculateMemorySize(32, MemoryAllowance.TxHashCacheSize) + LruCache<Keccak, Transaction>.CalculateMemorySize(4096, MemoryAllowance.MemPoolSize)) / 1000 / 1000}MB"
-                    .PadLeft(8));
+            AddNodeInfoEntryForTxPool();
 
             _transactions = new TxDistinctSortedPool(MemoryAllowance.MemPoolSize, comparer, logManager);
-            _persistentBroadcastTransactions = new TxDistinctSortedPool(MemoryAllowance.MemPoolSize, comparer, logManager);
-            _chainHeadInfoProvider.HeadChanged += OnHeadChange;
-            _ownTimer = new Timer(500);
-            _ownTimer.Elapsed += OwnTimerOnElapsed;
-            _ownTimer.AutoReset = false;
-            _ownTimer.Start();
+            _broadcaster = new TxBroadcaster(comparer, TimerFactory.Default, txPoolConfig, logManager);
+
+            _headInfo.HeadChanged += OnHeadChange;
+
+            _filterPipeline.Add(new NullHashTxFilter());
+            _filterPipeline.Add(new HashCacheBasedTxFilter(_hashCache));
```

Comment:
AlreadyKnownTxFilter

Reply:
I will skip opening Rider for this simple change but I agree with it so feel free to change later.